src/bin/ion/commands/generate/templates/java/class.templ

```diff
@@ -111,12 +125,18 @@ import java.io.IOException;
      *
      * This method does not close the writer after writing is complete.
      * The caller is responsible for closing the stream associated with the writer.
+     * This method skips writing a field when it's null.
```

Comment:
All the optional fields are represented as wrapper primitive classes now which removes the usage of `Optional.empty` and hence if a `null` is detected it would skip writing that field.